packages/wallets/src/components/WalletsResetMT5Password/__test__/WalletsResetMT5Password.spec.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { PropsWithChildren } from 'react';
+import { APIProvider, useTradingPlatformInvestorPasswordReset, useTradingPlatformPasswordReset } from '@deriv/api-v2';
+import { fireEvent, render, screen } from '@testing-library/react';
+import WalletsAuthProvider from '../../../AuthProvider';
+import { CFD_PLATFORMS } from '../../../features/cfd/constants';
+import useDevice from '../../../hooks/useDevice';
+import { ModalProvider } from '../../ModalProvider';
+import WalletsResetMT5Password from '../WalletsResetMT5Password';
+
+jest.mock('../../../hooks/useDevice');
+const mockUseDevice = useDevice as jest.MockedFunction<typeof useDevice>;
+
+jest.mock('../../ModalProvider', () => ({
+    ...jest.requireActual('../../ModalProvider'),
+    useModal: jest.fn(() => ({
+        hide: jest.fn(),
+        setModalOptions: jest.fn(),
+        show: jest.fn(),
+    })),
+}));
+
+jest.mock('@deriv/api-v2', () => ({
+    ...jest.requireActual('@deriv/api-v2'),
+    useTradingPlatformInvestorPasswordReset: jest.fn(() => ({
+        error: null,
+        isError: false,
+        isLoading: false,
+        isSuccess: false,
+        mutate: jest.fn(),
+    })),
+    useTradingPlatformPasswordReset: jest.fn(() => ({
+        error: null,
+        isError: false,
+        isLoading: false,
+        isSuccess: true,
+        mutate: jest.fn(),
+    })),
+}));
+
+jest.mock('../../WalletError', () => ({
+    __esModule: true,
+    default: () => <div>WalletError</div>,
+    ...jest.requireActual('../../WalletError'),
+}));
+
+jest.mock('../WalletSuccessResetMT5Password', () => ({
+    __esModule: true,
+    default: () => <div>WalletSuccessResetMT5Password</div>,
+    ...jest.requireActual('../WalletSuccessResetMT5Password'),
+}));
+
+const defaultProps = {
+    actionParams: 'test-action-params',
+    isInvestorPassword: false,
+    platform: CFD_PLATFORMS.MT5,
+    verificationCode: 'test-verification-code',
+};
+
+const wrapper = ({ children }: PropsWithChildren) => (
+    <APIProvider>
+        <WalletsAuthProvider>
+            <ModalProvider>{children}</ModalProvider>
+        </WalletsAuthProvider>
+    </APIProvider>
+);
+
+describe('WalletsResetMT5Password', () => {
```

Comment:
Seems like the test case does not check for the error and success modals. Could test cases be added to check for these i.e:
```
it('should render correct content for isChangePasswordError', () => {
        const mockMutate = jest.fn();
        mockUseDevice.mockReturnValue({
            isDesktop: true,
            isMobile: false,
            isTablet: false,
        });

        (useTradingPlatformPasswordReset as jest.Mock).mockReturnValue({
            error: 'Error',
            isError: true,
            isLoading: false,
            isSuccess: false,
            mutate: mockMutate,
        });

        render(<WalletsResetMT5Password {...defaultProps} />, { container: $root, wrapper });
        expect(screen.getAllByTestId('dt_modal_step_wrapper')[0]).toBeInTheDocument();
        expect(screen.getByText('WalletError')).toBeInTheDocument();
    });
```